core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -447,6 +447,12 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
         }
     })
 
+  def semiflatTap[C](f: B => F[C])(implicit F: Monad[F]): EitherT[F, A, B] =
+    semiflatMap(b => F.as(f(b), b))
+
+  def leftSemiflatTap[C](f: A => F[C])(implicit F: Monad[F]): EitherT[F, A, B] =
```

Comment:
What do we think about using `Unit` here to signify that this resulting type is going to be discarded? Might be a bit more cumbersome on the usage side, but it's just an extra `.void`

Reply:
We already have a `flatTap` on `FlatMap` that doesn't require `Unit`, and `tap`, `tapWith`, etc. on `Kleisli` don't constraint the output type to be `Unit`. My personal preference would be consistency—I think that plus slightly easier use outweighs a little extra expressivity in the types.

Reply:
@LukaJCB I first thought about it, but then saw that `FlatMap[F].flatTap` takes `A => F[B]` instead of `A => F[Unit]`, so I wanted to be consistent with existing signature.